--- vanillasrc/minecraft/net/minecraft/./util/ScreenShotHelper.java	2013-07-03 12:13:29.507343350 +0930
+++ modsrc/minecraft/net/minecraft/./util/ScreenShotHelper.java	2013-07-03 16:13:22.016821721 +0930
@@ -73,20 +73,9 @@
 
     private static File func_74290_a(File par0File)
     {
-        String s = dateFormat.format(new Date()).toString();
-        int i = 1;
-
-        while (true)
-        {
-            File file2 = new File(par0File, s + (i == 1 ? "" : "_" + i) + ".png");
-
-            if (!file2.exists())
-            {
-                return file2;
-            }
-
-            ++i;
-        }
+        Date now = new Date();
+        String basename = dateFormat.format(now).toString();
+        return watson.Controller.instance.getUniqueScreenshotFile(par0File, basename, now);
     }
 
     private static void func_74289_a(int[] par0ArrayOfInteger, int par1, int par2)
