--- vanillasrc/minecraft/net/minecraft/./client/gui/GuiChat.java	2013-08-04 09:13:43.692469714 +0930
+++ modsrc/minecraft/net/minecraft/./client/gui/GuiChat.java	2013-08-04 12:02:48.766033167 +0930
@@ -126,9 +126,9 @@
             {
                 this.mc.ingameGUI.getChatGUI().addToSentMessages(s);
 
-                if (!this.mc.handleClientCommand(s))
+                if (!clientcommands.mod_ClientCommands.getInstance().handleClientCommand(s))
                 {
-                    this.mc.thePlayer.sendChatMessage(s);
+                    this.mc.thePlayer.sendChatMessage(clientcommands.mod_ClientCommands.getInstance().executeChatCallbacks(s));
                 }
             }
 
@@ -372,4 +372,6 @@
     {
         return false;
     }
+
+    public GuiTextField getInputField() { return this.inputField; }    
 }
