--- vanillasrc/minecraft/net/minecraft/src/GuiChat.java	2013-05-03 14:27:32.225063870 +0930
+++ modsrc/minecraft/net/minecraft/src/GuiChat.java	2013-05-03 14:53:56.272948950 +0930
@@ -99,9 +99,9 @@
             {
                 this.mc.ingameGUI.getChatGUI().addToSentMessages(var3);
 
-                if (!this.mc.handleClientCommand(var3))
+                if (!mod_ClientCommands.getInstance().handleClientCommand(var3))
                 {
-                    this.mc.thePlayer.sendChatMessage(var3);
+                    this.mc.thePlayer.sendChatMessage(mod_ClientCommands.getInstance().executeChatCallbacks(var3));
                 }
             }
 
@@ -365,4 +365,6 @@
     {
         return false;
     }
+
+    public GuiTextField getInputField() { return this.inputField; }
 }
