--- vanillasrc/minecraft/net/minecraft/client/gui/GuiChat.java	2013-07-03 12:13:31.317342739 +0930
+++ modsrc/minecraft/net/minecraft/client/gui/GuiChat.java	2013-07-03 12:13:37.594340615 +0930
@@ -103,9 +103,9 @@
             {
                 this.mc.ingameGUI.getChatGUI().addToSentMessages(s);
 
-                if (!this.mc.handleClientCommand(s))
+                if (!clientcommands.mod_ClientCommands.getInstance().handleClientCommand(s))
                 {
-                    this.mc.thePlayer.sendChatMessage(s);
+                    this.mc.thePlayer.sendChatMessage(clientcommands.mod_ClientCommands.getInstance().executeChatCallbacks(s));
                 }
             }
 
@@ -369,4 +369,6 @@
     {
         return false;
     }
+    
+    public GuiTextField getInputField() { return this.inputField; }    
 }
