--- vanillasrc/minecraft/net/minecraft/src/GuiChat.java	2012-12-22 14:13:20.553969521 +1030
+++ modsrc/minecraft/net/minecraft/src/GuiChat.java	2012-12-22 14:22:33.412244918 +1030
@@ -99,9 +99,9 @@
             {
                 this.mc.ingameGUI.getChatGUI().addToSentMessages(var3);
 
-                if (!this.mc.handleClientCommand(var3))
+                if (!mod_ClientCommands.getInstance().handleClientCommand(var3))
                 {
-                    this.mc.thePlayer.sendChatMessage(var3);
+                    this.mc.thePlayer.sendChatMessage(mod_ClientCommands.getInstance().executeChatCallbacks(var3));
                 }
             }
 
@@ -365,4 +365,6 @@
     {
         return false;
     }
+
+    public GuiTextField getInputField() { return this.inputField; }
 }
