--- vanillasrc/minecraft/net/minecraft/src/NetClientHandler.java	2013-03-16 06:09:23.135431797 +1030
+++ modsrc/minecraft/net/minecraft/src/NetClientHandler.java	2013-03-16 06:41:52.634100971 +1030
@@ -697,7 +697,7 @@
 
     public void handleChat(Packet3Chat par1Packet3Chat)
     {
-        this.mc.ingameGUI.getChatGUI().printChatMessage(par1Packet3Chat.message);
+        watson.chat.ChatProcessor.getInstance().addChatToQueue(par1Packet3Chat.message);
         ModLoader.clientChat(par1Packet3Chat.message);
     }
 
