--- vanillasrc/minecraft/net/minecraft/src/NetClientHandler.java	2012-12-22 14:13:20.597963733 +1030
+++ modsrc/minecraft/net/minecraft/src/NetClientHandler.java	2012-12-22 14:22:33.415244523 +1030
@@ -680,7 +680,7 @@
 
     public void handleChat(Packet3Chat par1Packet3Chat)
     {
-        this.mc.ingameGUI.getChatGUI().printChatMessage(par1Packet3Chat.message);
+        watson.chat.ChatProcessor.getInstance().addChatToQueue(par1Packet3Chat.message);
         ModLoader.clientChat(par1Packet3Chat.message);
     }
 
