--- vanillasrc/minecraft/net/minecraft/src/NetClientHandler.java	2012-10-31 15:47:32.952700866 +1030
+++ modsrc/minecraft/net/minecraft/src/NetClientHandler.java	2012-10-31 19:29:39.805974074 +1030
@@ -686,7 +686,7 @@
 
     public void handleChat(Packet3Chat par1Packet3Chat)
     {
-        this.mc.ingameGUI.getChatGUI().printChatMessage(par1Packet3Chat.message);
+        watson.chat.ChatProcessor.getInstance().addChatToQueue(par1Packet3Chat.message);
         ModLoader.clientChat(par1Packet3Chat.message);
     }
 
