--- vanillasrc/minecraft/net/minecraft/src/NetClientHandler.java	2012-11-16 19:47:19.764532743 +1030
+++ modsrc/minecraft/net/minecraft/src/NetClientHandler.java	2012-11-16 20:17:57.346149034 +1030
@@ -683,7 +683,7 @@
 
     public void handleChat(Packet3Chat par1Packet3Chat)
     {
-        this.mc.ingameGUI.getChatGUI().printChatMessage(par1Packet3Chat.message);
+        watson.chat.ChatProcessor.getInstance().addChatToQueue(par1Packet3Chat.message);
         ModLoader.clientChat(par1Packet3Chat.message);
     }
 
